internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -626,6 +626,35 @@ func (r *BMACReconciler) findInstallationDiskID(devices []aiv1beta1.HostDisk, hi
 	return ""
 }
 
+// Finds the agents related to this ClusterDeployment
+//
+// The ClusterDeployment <-> agent relation is one-to-many.
+// This function returns all Agents whose ClusterDeploymentName name
+// matches this ClusterDeployment's name.
+func (r *BMACReconciler) findAgentsByClusterDeployment(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment) []*aiv1beta1.Agent {
+	agentList := aiv1beta1.AgentList{}
+	err := r.Client.List(ctx, &agentList, client.InNamespace(clusterDeployment.Namespace))
```

Comment:
Agents won't necessarily be in the same namespace as the ClusterDeployment. I think the normal use case will have them in different namespaces.

Reply:
I had a similar need (to get a list of agents that belong to a given clusterDeployment) for subsystem tests that I added for a different PR.
Ended up doing this:
https://github.com/openshift/assisted-service/pull/1606/files#diff-f2d3cebc87fae81285493762229b8e7ecac6ab614ea0d82950af55f303dcc3d5R195-R207

I wonder if there's a better way.

Reply:
I think that's a reasonable starting point.

An improvement could be to do this, maybe as a follow-up:
* have the Agent controller add a label that identifies to the assigned CD
* search Agents by label

The controller's client can be configured to index by label even in case that ends up being useful.

Reply:
Ok I will remove the namespace filter in this PR. Will add the search Agents by label in a follow-up PR.